### DISCUSSION TITLE
libfive compile size

### DISCUSSION THREAD

**Discussion by gsohler** [libfive/libfive#539] [General]
I was able to successfully build libfive for windows
but my program did not run because libfive.dll was missing.
my first idea to get libfive statically linked to my binary
but when i realized, its size is 422 MB, i thought  its still a good idea to keep it separate.

WHY IS it SO BIG ?
does include huge lookuptables to improve performance ?

**Comment by mkeeter** [comment]
On my machine, I see `libfive.dylib` at 2.2 MB:
```
➜  ls -lh build/libfive/src/libfive.dylib
-rwxr-xr-x  1 mkeeter  staff   2.2M Mar 24 08:22 build/libfive/src/libfive.dylib*
```

Perhaps you have unstripped debug symbols in `libfive.dll`?

**Comment by gsohler** [comment]
Hi Matt, this sounds like much better! ( i understand you are on a MAC BTW)
After all, do you konw if libfive  CMakeLists.txt has a hidden feature to compile statically ?
(and of course strip debug symbols ?)

**Reply by mkeeter** [reply]
I'm not sure what you mean by 'hidden feature'; everything is visible in the CMake files.

To compile a static library, you'd start by modifying [this line](https://github.com/libfive/libfive/blob/master/libfive/src/CMakeLists.txt#L5) to read `add_library(libfive STATIC`, then see what breaks from there; I'm not sure how to strip debug symbols on Windows.

**Comment by gsohler** [comment]
Hi Matt, after some reading I now understand add_library and target_link_libraries command.
I  was able to successfully statically link libfive  to openscad and nothing broke :)
However I try to avoid changing the CMakeLists.txt  in your github repo(because its not mine) and
when trying to automate things, only viable option  to me, is to patch your files once git has added it.
(but this is nasty)
A nice  option would be that you add an option in your CMakeLists.txt and made use of it with target_compile_defintions.
let me know ...
